packages/itwinui-react/src/core/Panels/Panels.tsx

```diff
@@ -0,0 +1,600 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import * as React from 'react';
+import {
+  Box,
+  cloneElementWithRef,
+  getWindow,
+  mergeEventHandlers,
+  SvgChevronLeft,
+  useInstance,
+  useMergedRefs,
+} from '../../utils/index.js';
+import type { PolymorphicForwardRefComponent } from '../../utils/index.js';
+import { IconButton } from '../Buttons/IconButton.js';
+import type { IconButtonProps } from '../Buttons/IconButton.js';
+import { Flex } from '../Flex/Flex.js';
+import { Text } from '../Typography/Text.js';
+import cx from 'classnames';
+import { panelAnimationReducer, PanelsInstanceProvider } from './helpers.js';
+import type {
+  CurrentPanelId,
+  PanelAnimationState,
+  PanelsInstance,
+  TriggerMapEntry,
+} from './helpers.js';
+
+type PanelsWrapperProps = {
+  /**
+   * The initialPanel that is displayed.
+   */
+  initialActiveId: string;
+  /**
+   * Pass a value to enter controlled mode. The value object has a few fields:
+   * - `id: string` - The panel to show.
+   * - `direction?: 'prev' | 'next' | 'undefined'`:
+   *   - animates to the new panel in the specified direction or transitions instantly if `undefined`.
+   *   - moves focus to the new panel when `next` or `undefined` and to the trigger when `prev`.
+   */
+  activeId?: CurrentPanelId;
+  /** Useful for controlled mode */
+  onActiveIdChange?: (newActiveId: CurrentPanelId) => void;
+  children: React.ReactNode;
+  /**
+   * Pass an instance created by `useInstance` to control the panels imperatively.
+   *
+   * @example
+   * const panels = Panels.useInstance();
+   * <Panels instance={panels} />
+   */
+  instance?: PanelsInstance;
+  /**
+   * To customize the panel transition animations, pass one of the following:
+   * - `number`: Duration in milliseconds. Overrides the default `animationOptions`.
+   * - `object`: For more customization. Merges with the default `animationOptions`.
+   * - `null`: Disables animations completely.
+   *
+   * @see `HTMLElement.animate`
+   */
+  animationOptions?: Parameters<HTMLElement['animate']>[1] | null;
```

Comment:
I think we can also leave out animationOptions from the initial implementation. We can revisit based on user needs.